src/js/pannellum.js

```diff
@@ -2117,13 +2136,23 @@ function constrainHfov(hfov) {
         // Don't change view if bounds don't make sense
         console.log('HFOV bounds do not make sense (minHfov > maxHfov).')
         return config.hfov;
-    } if (hfov < minHfov) {
-        return minHfov;
+    }
+    var newHfov = config.hfov;
+    if (hfov < minHfov) {
+        newHfov = minHfov;
     } else if (hfov > config.maxHfov) {
-        return config.maxHfov;
+        nwHfov = config.maxHfov;
     } else {
-        return hfov;
+        newHfov = hfov;
+    }
+    if (renderer) {
+        // prevent showing background (empty space) on top or bottom by adapting newHfov
+        newHfov = Math.min(newHfov,
+                           Math.atan(Math.tan((config.maxPitch - config.minPitch) / 360 * Math.PI) /
+                                     renderer.getCanvas().height * renderer.getCanvas().width)
```

Comment:
It would be better to assign the result of `getCanvas()` to a variable instead of calling it twice.

Reply:
Indeed